File: mobile/test/rad_mark_test.dart
import 'package:aetherride_mobile/domain/bike.dart';
import 'package:aetherride_mobile/domain/component.dart';
import 'package:aetherride_mobile/domain/garage/die_box.dart';
import 'package:aetherride_mobile/domain/garage/rad_mark.dart';
import 'package:flutter_test/flutter_test.dart';

void main() {
  test('Rad-Stand marks match the web mapping', () {
    expect(radMarkForItem(DieBoxItemId.pressureUnknown), 'pressure');
    expect(radMarkForItem(DieBoxItemId.sagUnknown), 'sag');
    expect(radMarkForItem(DieBoxItemId.lightsMissing), 'lights');
    expect(radMarkForItem(DieBoxItemId.dueCare), 'care');
    expect(radMarkForItem(DieBoxItemId.setActive), 'stand');
    expect(radMarkForItem(DieBoxItemId.parkTrail), 'setup');

    expect(radMarkForReadiness(DieBoxReadiness.ready), 'ready');
    expect(radMarkForReadiness(DieBoxReadiness.almost), 'almost');
    expect(radMarkForReadiness(DieBoxReadiness.unknown), 'unknown');

    expect(radMarkForChip('Druck'), 'pressure');
    expect(radMarkForChip('Licht'), 'lights');
    expect(radMarkForChip('Ausweis'), 'identity');
    expect(radMarkForChip('140/150 mm'), 'travel');
    expect(radMarkForChip('29"'), 'pressure');
    expect(radMarkForChip('700c'), 'pressure');

    expect(radMarkForSlot(ComponentSlot.tireFront), 'pressure');
    expect(radMarkForSlot(ComponentSlot.chain), 'chain');
    expect(radMarkForSlot(ComponentSlot.brakeRear), 'brakes');
    expect(radMarkForSlot(ComponentSlot.battery), 'battery');
    expect(radMarkForSlot(ComponentSlot.handlebar), 'cockpit');
    expect(radMarkForSlot(ComponentSlot.frame), 'parts');
  });

  test('Silhouette assets sit on the stand, hiking is not a dashed bike', () {
    expect(
      radSilhouetteAsset(Bike(id: 'g', name: 'G', category: BikeCategory.gravel)),
      'assets/garage/silhouettes/gravel.svg',
    );
    expect(
      radSilhouetteAsset(Bike(id: 'e', name: 'E', category: BikeCategory.emtb)),
      'assets/garage/silhouettes/emtb.svg',
    );
    expect(
      radSilhouetteAsset(
        Bike(id: 'h', name: 'Tour', category: BikeCategory.hiking),
      ),
      radHikingAsset,
    );
    expect(radHikingAsset, isNot(radNoPhotoAsset));
  });
}

File: src/lib/garage/radMark.test.ts
/**
 * FlowLine Rad-Stand marks — Ausführen: npx tsx src/lib/garage/radMark.test.ts
 */
import assert from "node:assert/strict";
import { existsSync } from "node:fs";
import { join } from "node:path";
import {
  RAD_EMPTY_STAND,
  RAD_EMPTY_STAND_MARK,
  RAD_MARKS,
  RAD_MARK_SRC,
  RAD_NO_PHOTO,
  RAD_STAND_GROUND,
  RAD_STAND_HEADER,
  radMarkForChip,
  radMarkForItem,
  radMarkForMeasure,
  radMarkForReadiness,
  radMarkForSlot,
  radSilhouetteSrc,
} from "./radMark";

const root = join(import.meta.dirname, "../../..");
for (const name of RAD_MARKS) {
  assert.match(RAD_MARK_SRC[name], /^\/garage\/glyph-.+\.svg$/, name);
  assert.ok(
    existsSync(join(root, "public", RAD_MARK_SRC[name].slice(1))),
    `${name} svg`
  );
}
for (const src of [
  RAD_STAND_HEADER,
  RAD_STAND_GROUND,
  RAD_EMPTY_STAND,
  RAD_EMPTY_STAND_MARK,
  RAD_NO_PHOTO,
]) {
  assert.ok(existsSync(join(root, "public", src.slice(1))), src);
}

assert.equal(radMarkForItem("pressureUnknown"), "pressure");
assert.equal(radMarkForItem("sagUnknown"), "sag");
assert.equal(radMarkForItem("lightsMissing"), "lights");
assert.equal(radMarkForItem("dueCare"), "care");
assert.equal(radMarkForItem("setActive"), "stand");
assert.equal(radMarkForItem("parkTrail"), "setup");

assert.equal(radMarkForReadiness("ready"), "ready");
assert.equal(radMarkForReadiness("almost"), "almost");
assert.equal(radMarkForReadiness("unknown"), "unknown");

assert.equal(radMarkForChip("Druck"), "pressure");
assert.equal(radMarkForChip("Licht"), "lights");
assert.equal(radMarkForChip("Ausweis"), "identity");
assert.equal(radMarkForChip("140/150 mm"), "travel");
assert.equal(radMarkForChip('29"'), "pressure");
assert.equal(radMarkForChip("700c"), "pressure");
assert.equal(radMarkForChip("27.5\""), "pressure");

assert.equal(radMarkForSlot("tire_front"), "pressure");
assert.equal(radMarkForSlot("chain"), "chain");
assert.equal(radMarkForSlot("brake_rear"), "brakes");
assert.equal(radMarkForSlot("battery"), "battery");
assert.equal(radMarkForSlot("handlebar"), "cockpit");
assert.equal(radMarkForSlot("frame"), "parts");

assert.equal(radMarkForMeasure("pressure"), "pressure");
assert.equal(radMarkForMeasure("sag"), "sag");
assert.equal(radMarkForMeasure("travel"), "travel");
assert.match(radSilhouetteSrc({ category: "gravel" }), /gravel\.svg$/);
assert.match(radSilhouetteSrc({ category: "emtb" }), /emtb\.svg$/);
assert.match(radSilhouetteSrc({ category: "hiking" }), /hiking\.svg$/);
assert.notEqual(radSilhouetteSrc({ category: "hiking" }), RAD_NO_PHOTO);
assert.ok(
  existsSync(join(root, "public/garage/silhouettes/hiking.svg")),
  "hiking svg"
);

console.log("radMark.test.ts ok");
